### PR TITLE
add error code endpoint using error as enum

### PR DESCRIPTION
**Pull request by inkhey**
related to #1006

### CODE REVIEW

File: backend/tracim_backend/tests/functional/test_account.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import requests
 import transaction
 
-from tracim_backend import error
+from tracim_backend.error import ErrorCodes as error
```
Comment (lebouquetin) [inline backend/tracim_backend/tests/functional/test_account.py]: do not use `import as`

File: backend/tracim_backend/tests/functional/test_comments.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 import transaction
 
-from tracim_backend import error
+from tracim_backend.error import ErrorCodes as error
```
Comment (lebouquetin) [inline backend/tracim_backend/tests/functional/test_comments.py]: do not use `import as`

File: backend/tracim_backend/tests/functional/test_comments.py [hunk 3]
```diff
@@ -11,7 +11,7 @@
 from tracim_backend.models.revision_protection import new_revision
 from tracim_backend.models.setup_models import get_tm_session
 from tracim_backend.tests import FunctionalTest
-from tracim_backend import error
+from tracim_backend.error import ErrorCodes as error
```
Comment (lebouquetin) [inline backend/tracim_backend/tests/functional/test_comments.py]: do not use `import as`

File: backend/tracim_backend/tests/functional/test_contents.py [hunk 4]
```diff
@@ -16,7 +16,7 @@
 from depot.io.utils import FileIntent
 from PIL import Image
 
-from tracim_backend import error
+from tracim_backend.error import ErrorCodes as error
```
Comment (lebouquetin) [inline backend/tracim_backend/tests/functional/test_contents.py]: do not use `import as`

File: backend/tracim_backend/tests/functional/test_reset_password.py [hunk 5]
```diff
@@ -4,7 +4,7 @@
 from freezegun import freeze_time
 from tracim_backend.models.auth import User
 from tracim_backend.models.auth import AuthType
-from tracim_backend import error
+from tracim_backend.error import ErrorCodes as error
```
Comment (lebouquetin) [inline backend/tracim_backend/tests/functional/test_reset_password.py]: do not use `import as`

File: backend/tracim_backend/tests/functional/test_session.py [hunk 6]
```diff
@@ -3,7 +3,7 @@
 import transaction
 from freezegun import freeze_time
 from tracim_backend.models.auth import User
-from tracim_backend import error
+from tracim_backend.error import ErrorCodes as error
```
Comment (lebouquetin) [inline backend/tracim_backend/tests/functional/test_session.py]: do not use `import as`

File: backend/tracim_backend/tests/functional/test_user.py [hunk 7]
```diff
@@ -10,7 +10,7 @@
 
 from tracim_backend import AuthType
 from tracim_backend.models.auth import User
-from tracim_backend import error
+from tracim_backend.error import ErrorCodes as error
```
Comment (lebouquetin) [inline backend/tracim_backend/tests/functional/test_user.py]: do not use `import as`

File: backend/tracim_backend/tests/functional/test_workspaces.py [hunk 8]
```diff
@@ -9,7 +9,7 @@
 
 
 from tracim_backend import AuthType
-from tracim_backend import error
+from tracim_backend.error import ErrorCodes as error
```
Comment (lebouquetin) [inline backend/tracim_backend/tests/functional/test_workspaces.py]: do not use `import as`

File: backend/tracim_backend/lib/core/system.py [hunk 9]
```diff
@@ -27,3 +30,9 @@ def get_config(self) -> ConfigModel:
             email_notification_activated=self._config.EMAIL_NOTIFICATION_ACTIVATED,  # nopep8
             new_user_invitation_do_notify=self._config.NEW_USER_INVITATION_DO_NOTIFY,  # nopep8
         )
+
+    def get_error_codes(self) -> typing.List[ErrorCodeModel]:
+        error_codes = []
+        for error_code in ErrorCodes:
```
[the remark this answers is not in the document]
Comment (inkhey) [reply backend/tracim_backend/lib/core/system.py]: Not sure to understand. foreach is the standard way to get all enum of enum type.

File: backend/tracim_backend/error.py [hunk 10]
```diff
@@ -7,68 +8,69 @@
...
-WORKSPACE_NOT_FOUND = 1002
-CONTENT_NOT_FOUND = 1003
-PARENT_NOT_FOUND = 1004
-USER_ROLE_NOT_FOUND = 1005
-CONTENT_TYPE_NOT_EXIST = 1006
-# Preview Based
-UNAVAILABLE_PREVIEW_TYPE = 1011
-PAGE_OF_PREVIEW_NOT_FOUND = 1012
-UNAIVALABLE_PREVIEW = 1013
+class ErrorCodes(IntEnum):
```
[the remark this answers is not in the document]
Comment (inkhey) [reply backend/tracim_backend/error.py]: done

File: backend/tracim_backend/tests/functional/test_account.py (hunk 1 above)
Comment (inkhey) [reply backend/tracim_backend/tests/functional/test_account.py]: done

File: backend/tracim_backend/tests/functional/test_comments.py (hunk 2 above)
Comment (inkhey) [reply backend/tracim_backend/tests/functional/test_comments.py]: done

File: backend/tracim_backend/tests/functional/test_contents.py (hunk 4 above)
Comment (inkhey) [reply backend/tracim_backend/tests/functional/test_contents.py]: done

File: backend/tracim_backend/tests/functional/test_reset_password.py (hunk 5 above)
Comment (inkhey) [reply backend/tracim_backend/tests/functional/test_reset_password.py]: done

File: backend/tracim_backend/tests/functional/test_session.py (hunk 6 above)
Comment (inkhey) [reply backend/tracim_backend/tests/functional/test_session.py]: done

File: backend/tracim_backend/tests/functional/test_workspaces.py (hunk 8 above)
Comment (inkhey) [reply backend/tracim_backend/tests/functional/test_workspaces.py]: done

File: backend/tracim_backend/tests/functional/test_user.py (hunk 7 above)
Comment (inkhey) [reply backend/tracim_backend/tests/functional/test_user.py]: done.

File: backend/tracim_backend/tests/functional/test_comments.py (hunk 3 above)
Comment (inkhey) [reply backend/tracim_backend/tests/functional/test_comments.py]: done.

File: backend/tracim_backend/lib/core/system.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (lebouquetin) [reply backend/tracim_backend/lib/core/system.py]: ok, i didnt know that feature

**Pull request merged**